components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -354,6 +354,52 @@ public API getAPIbyUUID(String uuid) throws APIManagementException {
         }
     }
 
+    /**
+     * Get minimal details of API by registry artifact id
+     *
+     * @param uuid  Registry artifact id
+     * @return API of the provided artifact id
+     * @throws APIManagementException
+     */
+    public API getAPIInformationByUUID(String uuid, String requestedTenantDomain) throws APIManagementException {
+        try {
+            Registry registry;
+            if (!requestedTenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                int id = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                        .getTenantId(requestedTenantDomain);
+                registry = ServiceReferenceHolder.getInstance().
+                        getRegistryService().getGovernanceSystemRegistry(id);
+            } else {
+                if (this.tenantDomain != null && !this.tenantDomain
+                        .equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                    // at this point, requested tenant = carbon.super but logged in user is anonymous or tenant
+                    registry = ServiceReferenceHolder.getInstance().
+                            getRegistryService().getGovernanceSystemRegistry(MultitenantConstants.SUPER_TENANT_ID);
+                } else {
+                    // both requested tenant and logged in user's tenant are carbon.super
+                    registry = this.registry;
+                }
+            }
+            GenericArtifactManager artifactManager = APIUtil.getArtifactManager(registry, APIConstants.API_KEY);
+
+            GenericArtifact apiArtifact = artifactManager.getGenericArtifact(uuid);
+            if (apiArtifact != null) {
+                return APIUtil.getAPIInformation(apiArtifact, registry);
+            } else {
+                String errorMessage =
+                        "Failed to get API. API artifact corresponding to artifactId " + uuid + " does not exist";
+                handleException(errorMessage);
+                return null;
+            }
+        } catch (RegistryException e) {
+            handleException("Failed to get API", e);
```

Comment:
Added more descriptive error messaged in #1581 (https://github.com/wso2/carbon-apimgt/commit/222c4a73ac87d98718629fb551a500e42e432b83)